roles/openshift_master/tasks/systemd_units.yml

```diff
@@ -38,6 +38,11 @@
   when: create_ha_unit_files | changed
 # end workaround for missing systemd unit files
 
+- name: Preserve Master API Proxy Config options
+  command: grep PROXY /etc/sysconfig/{{ openshift.common.service_type }}-master-api
+  register: master_api_proxy
+  when: openshift.master.ha is defined and openshift.master.ha | bool and openshift_master_cluster_method == "native"
```

Comment:
I think this particular implementation keeps us from having to parse them. If it's there and we don't have these variables defined, either on disk or from inventory, then we just restore them exactly as they are after making whatever other changes we need to make.